app/src/main/java/com/globant/marvelmvvm/viewmodel/AllCharactersViewModel.kt

```diff
@@ -1,7 +1,42 @@
 package com.globant.marvelmvvm.viewmodel
 
+import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.viewModelScope
 import com.globant.marvelmvvm.contract.AllCharactersContract
+import com.globant.marvelmvvm.data.entity.Character
+import com.globant.marvelmvvm.util.Data
+import com.globant.marvelmvvm.util.Event
+import com.globant.marvelmvvm.util.Result
+import com.globant.marvelmvvm.util.Status
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 
-class AllCharactersViewModel: ViewModel(), AllCharactersContract.ViewModel {
+class AllCharactersViewModel(private val model: AllCharactersContract.Model) : ViewModel(), AllCharactersContract.ViewModel {
+
+    var mutableMainState: MutableLiveData<Event<Data<List<Character>>>> = MutableLiveData()
+    override val mainState: LiveData<Event<Data<List<Character>>>>
+        get() = mutableMainState
```

Comment:
you can use a simple assignment here: 
override val mainState: LiveData<Event<Data<List<Character>>>> = mutableMainState

Reply:
Done!